src/components/Checkbox/index.test.tsx

```diff
@@ -15,57 +15,70 @@ describe('Checkbox component', () => {
     userEvent.click(input);
     expect(input).not.toBeChecked();
     expect(screen.getByRole('checkbox')).toHaveAttribute('disabled');
-    expect(container.querySelector('.fdsCheckable')).toHaveClass(
-      'fdsCheckable--disabled'
+    expect(container.querySelector('.fdsCheckableCheckbox')).toHaveClass(
+      'fdsCheckableCheckbox--disabled'
     );
   });
 
   it('fires change callback when checking', () => {
-    const changeFn = jest.fn();
+    const mockOnChange = jest.fn();
     render(
-      <Checkbox label="change callback" name="check-callback-test" onChange={changeFn} />
+      <Checkbox
+        label="change callback"
+        name="check-callback-test"
+        onChange={mockOnChange}
+      />
     );
     const input = screen.getByLabelText('change callback', { exact: false });
-    expect(changeFn).not.toHaveBeenCalled();
+    expect(mockOnChange).not.toHaveBeenCalled();
     userEvent.click(input);
     expect(input).toBeChecked();
     userEvent.type(input, 'change callback');
-    expect(changeFn).toHaveBeenCalled();
+    expect(mockOnChange).toHaveBeenCalled();
   });
 
   it('fires change callback when unchecking', () => {
-    const changeFn = jest.fn();
+    const mockOnChange = jest.fn();
     render(
       <Checkbox
         defaultChecked
         label="unchecking callback"
         name="uncheck-callback-test"
-        onChange={changeFn}
+        onChange={mockOnChange}
       />
     );
 
     const input = screen.getByLabelText('unchecking callback', { exact: false });
-    expect(changeFn).not.toHaveBeenCalled();
+    expect(mockOnChange).not.toHaveBeenCalled();
     userEvent.click(input);
     expect(input).not.toBeChecked();
     userEvent.type(input, 'unchecking callback');
-    expect(changeFn).toHaveBeenCalled();
+    expect(mockOnChange).toHaveBeenCalled();
   });
 
   it('allows users to supply their own id', () => {
+    const mockOnChange = jest.fn();
     render(
       <>
         <Checkbox
           id="checkbox-id"
           label="Component label"
           showLabel={false}
           name="multiple labels"
+          onChange={mockOnChange}
         />
         <label htmlFor="checkbox-id">User label</label>
       </>
     );
 
-    expect(screen.getByLabelText('Component label')).toBeInTheDocument();
-    expect(screen.getByLabelText('User label')).toBeInTheDocument();
+    expect(screen.queryByLabelText('Component label')).not.toBeInTheDocument();
+    expect(screen.getByText('User label')).toBeInTheDocument();
+
+    const input = screen.getByText('User label', { exact: false });
+    expect(mockOnChange).not.toHaveBeenCalled();
+    userEvent.click(input);
+    expect(input).not.toBeChecked();
+    userEvent.type(input, 'User label');
+    expect(mockOnChange).toHaveBeenCalled();
```

Comment:
it couldn't find this as a label for some reason - I made a story to test it out and clicking on this user provided label does check/uncheck the checkbox but i did not see the action in the actions tab. I added the mockOnChange here to ensure that the onChange would still be called when user provides the label

Reply:
@David-Cohen35 k, I would just change the one in the test to be a span.

Reply:
htmlFor is not assignable to span so it wouldnt be tied to the checkbox

Reply:
if we leave it as is (with getByText instead of getByLabelText) the test is still checking that we can see the user supplied label and that the onChange callback is being fired when that label is clicked. What is the danger in leaving it like this?

Reply:
ah, you're right. Yeah we can move forward with this as is, good push back.